frontend/src/components/common/buttons/Button.tsx

```diff
@@ -0,0 +1,235 @@
+import classnames from 'classnames';
+import React, { CSSProperties, forwardRef, MouseEventHandler } from 'react';
+import BootstrapButton, { ButtonProps as BootstrapButtonProps } from 'react-bootstrap/Button';
+import Spinner from 'react-bootstrap/Spinner';
+import styled from 'styled-components';
+
+export interface ButtonProps extends BootstrapButtonProps {
+  /** Adds a custom class to the button element of the <Button> component */
+  className?: string;
+  /** Allow for direct style overrides */
+  style?: CSSProperties;
+  /** Sets an icon before the text. Can be any icon from Evergreen or a custom element. */
+  icon?: React.ReactNode;
+  /** When true, the button is disabled. isLoading also sets the button to disabled. */
+  disabled?: boolean;
+  /** Button click handler */
+  onClick?: MouseEventHandler<any>;
+  /** Display a spinner when the form is being submitted */
+  showSubmitting?: boolean;
+  /** if true and showSubmitting is true, display the spinner */
+  isSubmitting?: boolean;
+  /** default button text value */
+  defaultValue?: string | number | string[];
+  /** overwrite type from React.HTMLAttributes */
+  'aria-relevant'?: 'text' | 'all' | 'additions' | 'additions text' | 'removals';
+}
+
+/**
+ * Buttons allow users to take actions, and make choices, with a single tap.
+ * Buttons are used primarily for actions, such as “Add”, “Close”, “Cancel”, or “Save”.
+ * Plain buttons, which look similar to links, are used for less important or
+ * less commonly used actions, such as "View more details".
+ */
+export const Button = forwardRef<typeof StyledButton, ButtonProps>((props, ref) => {
+  const { showSubmitting, isSubmitting, disabled, icon, children, className, ...rest } = props;
+
+  const classes = classnames({
+    Button: true,
+    'Button--disabled': disabled,
+    'Button--icon-only': (children === null || children === undefined) && icon,
+    [className!]: className,
+  });
+
+  return (
+    <StyledButton ref={ref} className={classes} disabled={disabled} {...rest}>
+      {icon && <div className="Button__icon">{icon}</div>}
+      {children && <div className="Button__value">{children}</div>}
+      {showSubmitting && isSubmitting && (
+        <Spinner
+          animation="border"
+          size="sm"
+          role="status"
+          as="span"
+          style={{ marginLeft: '.8rem', padding: '.8rem' }}
+        />
+      )}
+    </StyledButton>
+  );
+});
+
+const StyledButton = styled(BootstrapButton)`
+  &.btn {
```

Comment:
Instead of having the css change based on the variant, what do you think about returning different styles that inherit from bootstrap? Just asking to see if would make the maintenance of the buttons easier